zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,402 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from textwrap import dedent
+
 import bcolz
+from bcolz import ctable
+import numpy as np
+from os.path import join
 import json
 import os
 import pandas as pd
 
 MINUTES_PER_DAY = 390
-
+DEFAULT_EXPECTEDLEN = 390 * 252 * 15
+OHLC_RATIO = 1000
 METADATA_FILENAME = 'metadata.json'
 
 
+class BcolzMinuteOverlappingData(Exception):
+    pass
+
+
+def _calc_minute_index(market_opens, minutes_per_day):
+    minutes = np.zeros(len(market_opens) * minutes_per_day,
+                       dtype='datetime64[ns]')
+    deltas = np.arange(0, minutes_per_day, dtype='timedelta64[m]')
+    for i, market_open in enumerate(market_opens):
+        start = market_open.asm8
+        minute_values = start + deltas
+        start_ix = minutes_per_day * i
+        end_ix = start_ix + minutes_per_day
+        minutes[start_ix:end_ix] = minute_values
+    return pd.to_datetime(minutes, utc=True, box=True)
+
+
+def _sid_subdir_path(sid):
+    """
+    Format subdir path to limit the number directories in any given
+    subdirectory to 100.
+
+    The number in each directory is designed to support at least 100000
+    equities.
+
+    Parameters:
+    -----------
+    sid - int
+        Asset identifier.
+
+    Returns:
+    --------
+    string
```

Comment:
The format for this is usually the same as the input.  For example. `np.array` does:

```
Returns
-------
out : ndarray
    An array object satisfying the specified requirements.
```